src/netif.c

```diff
@@ -1770,87 +1848,128 @@ static int get_default_gateway(const struct ip_setting_handler *handler, struct
     return 0;
 }
 
+
+static size_t netif_count_rw_handlers()
+{
+  const struct ip_setting_handler *handler = &handlers[0];
+  size_t count = 0;
+
+  for(; handler->name != NULL; handler++) {
```

Comment:
Can we avoid doing 
```
for(;
```
There is nothing wrong with it, but it does cause most C people to stop and re-read the code.

something like this?
```
for(count = 0; handler->name != NULL; handler++)
```

Reply:
done...